yt_dlp/extractor/tencent.py

```diff
@@ -0,0 +1,367 @@
+        }
+
+        return self._search_json(r'QZOutputJson=', self._download_webpage(
+            self._API_URL, video_id, query=query), 'api_response', video_id)
+
+    def _extract_video_formats_and_subtitles(self, api_response, video_id):
+        video_response = api_response['vl']['vi'][0]
+        video_width, video_height = video_response.get('vw'), video_response.get('vh')
+
+        formats, subtitles = [], {}
+        for video_format in video_response['ul']['ui']:
+            if video_format.get('hls'):
+                fmts, subs = self._extract_m3u8_formats_and_subtitles(
+                    video_format['url'] + video_format['hls']['pt'], video_id, 'mp4', fatal=False)
+                for f in fmts:
+                    f.update({'width': video_width, 'height': video_height})
+
+                formats.extend(fmts)
+                self._merge_subtitles(subs, target=subtitles)
+            else:
+                formats.append({
+                    'url': f'{video_format["url"]}{video_response["fn"]}?vkey={video_response["fvkey"]}',
+                    'width': video_width,
+                    'height': video_height,
+                    'ext': 'mp4',
+                })
+
+        return formats, subtitles
+
+    def _extract_video_native_subtitles(self, api_response, subtitles_format):
+        subtitles = {}
+        for subtitle in traverse_obj(api_response, ('sfl', 'fi')) or ():
+            subtitles.setdefault(subtitle['lang'].lower(), []).append({
+                'url': subtitle['url'],
+                'ext': subtitles_format,
+                'protocol': 'm3u8_native' if determine_ext(subtitle['url']) == 'm3u8' else 'http',
+            })
+
+        return subtitles
+
+    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
+        formats, subtitles = [], {}
+        for video_format, subtitle_format, video_quality in (
+                # '': 480p, 'shd': 720p, 'fhd': 1080p
+                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
+            api_response = self._get_video_api_response(
+                url, video_id, series_id, subtitle_format, video_format, video_quality)
+
+            if api_response.get('em') != 0 and api_response.get('exem') != 0:
+                if '您所在区域暂无此内容版权' in api_response.get('msg'):
+                    self.raise_geo_restricted()
+                raise ExtractorError(f'Tencent said: {api_response.get("msg")}')
+
+            fmts, subs = self._extract_video_formats_and_subtitles(api_response, video_id)
+            native_subtitles = self._extract_video_native_subtitles(api_response, subtitle_format)
+
+            formats.extend(fmts)
+            self._merge_subtitles(subs, native_subtitles, target=subtitles)
+
+        self._sort_formats(formats)
+        return formats, subtitles
+
+    def _get_clean_title(self, title):
+        return re.sub(r'\s*[_\-]\s*(?:Watch online|腾讯视频|(?:高清)?1080P在线观看平台).*?$', '', title)
```

Comment:
```suggestion
        return re.sub(
            r'\s*[_\-]\s*(?:Watch online|腾讯视频|(?:高清)?1080P在线观看平台).*?$',
            '', title or '').strip() or None
```

Reply:
`strip_or_none()` ?

Reply:
that will return `''` instead of `None` when `title=None`